src/raft/recv_install_snapshot.c

```diff
@@ -462,16 +484,14 @@ static void follower_sent_cb(struct sender *s, int rc)
 		return;
 	}
 
-	sm_move(&rpc->sm, RPC_SENT);
 	follower_tick(follower, M_MSG_SENT);
 }
 
 static bool is_a_trigger_leader(const struct leader *leader, const struct raft_message *incoming)
 {
 	(void)leader;
-	/* Leader has no triggers on sending a message, it reacts when follower
-	 * replies */
-	return incoming != M_MSG_SENT;
```

Comment:
[Question for reviewer]: This was the other change that I am not sure of. I tried to make the way we handle messages more uniform. We react to `M_TIMEOUT` and `M_WORK_DONE` but we didn't use `M_MSG_SENT` we used to update the state of the rpc in the callback only in that case (from SENT to REPLIED we did it in the main loop). With this change we react to `M_MSG_SENT` and update the state in `rcp_tick`.

This also has the side-effect of being able to create the timeout transparently in the main loop.

Reply:
This change makes sense to me but do we still need `is_a_trigger_leader` afterward, other than for symmetry with `is_a_trigger_follower`?

Reply:
Not in this patch but the idea is that we will filter the messages to only get the ones we care about in that stage. That will come in a later patch when we do deduplication, react to unexpected messages, etc. So for now it is pending implementation.